internal-webapps/org.wso2.bfsi.consent.management.endpoint/src/main/java/org/wso2/bfsi/consent/management/endpoint/api/ConsentValidationEndpoint.java

```diff
@@ -0,0 +1,197 @@
+import org.wso2.bfsi.consent.management.service.impl.ConsentCoreServiceImpl;
+
+import java.net.URISyntaxException;
+import java.text.ParseException;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.TreeMap;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response;
+
+/**
+ * ConsentValidationEndpoint.
+ *
+ * This specifies a REST API for consent validation to be used at consent enforcement of resource
+ * retrieval/submission requests.
+ */
+@SuppressFBWarnings("JAXRS_ENDPOINT")
+// Suppressed content - Endpoints
+// Suppression reason - False Positive : These endpoints are secured with access control
+// as defined in the IS deployment.toml file
+// Suppressed warning count - 1
+@Path("/validate")
+public class ConsentValidationEndpoint {
+
+    private static final Log log = LogFactory.getLog(ConsentValidationEndpoint.class);
+    private static final ConsentCoreServiceImpl consentCoreService = new ConsentCoreServiceImpl();
+    private static ConsentValidator consentValidator = null;
+    private static String requestSignatureAlias;
+
+    public ConsentValidationEndpoint() {
+
+        if (consentValidator == null) {
+            initializeConsentValidator();
+        }
+    }
+
+    private static void initializeConsentValidator() {
+
+        ConsentValidateBuilder consentValidateBuilder = ConsentExtensionExporter.getConsentValidateBuilder();
+
+        if (consentValidateBuilder != null) {
+            consentValidator = consentValidateBuilder.getConsentValidator();
+            requestSignatureAlias = consentValidateBuilder.getRequestSignatureAlias();
+            log.info(String.format("Consent validator %s initialized",
+                    consentValidator.getClass().getName().replaceAll("\n\r", "")));
+        }
+
+        if (consentValidator == null) {
+            log.warn("Consent validator is null");
+        }
+    }
+
+    /**
+     * Validate by sending consent data.
+     */
+    @POST
+    @Path("/validate")
+    @Consumes({"application/jwt; charset=utf-8"})
+    @Produces({"application/json; charset=utf-8"})
+    public Response validate(@Context HttpServletRequest request, @Context HttpServletResponse response) {
+
+        String payload = ConsentUtils.getStringPayload(request);
+        JSONObject requestData;
+
+        if (ConsentUtils.getConsentJWTPayloadValidatorConfigEnabled()) {
+            try {
+                ConsentExtensionUtils.validateJWTSignatureWithPublicKey(payload, requestSignatureAlias);
+                requestData = new JSONObject(Objects
+                        .requireNonNull(CommonUtils.decodeRequestJWT(payload, "body")).toJSONString());
```

Comment:
```suggestion
                        .requireNonNull(CommonUtils.decodeRequestJWT(payload, ConsentExtensionConstants.BODY)).toJSONString());
```

Reply:
Fixed in https://github.com/wso2-extensions/bfsi-consent-management/pull/9/commits/7640a3fd126e5e00ce693961fdc6c5cd28816bb9